snuba/request/request_settings.py

```diff
@@ -119,6 +129,12 @@ def get_rate_limit_params(self) -> Sequence[RateLimitParameters]:
     def add_rate_limit(self, rate_limit_param: RateLimitParameters) -> None:
         self.__rate_limit_params.append(rate_limit_param)
 
+    def get_resource_quota(self) -> Optional[ResourceQuota]:
+        return self.__resource_quota
+
+    def add_resource_quota(self, quota: ResourceQuota) -> None:
```

Comment:
I would change this to `set_resource_quota` since you are overriding any previous settings.